src/store.rs

```diff
@@ -2435,6 +2456,13 @@ impl<F: LurkField> Store<F> {
             let name_str = self.get_str(name).unwrap();
 
             let (prev_full_name, full_name) = if let Some(x) = full_name_acc.clone() {
+                // We need to quote the path segment when constructing the incremental full-name.
+                // This is because when symbols are interned by full-name, the canonical name,
+                // which includes any necessary quoting, is used.
+                //
+                // Eventually, we will remove symbol lookup by full-name, simplifying this.
+                let name = crate::parser::maybe_quote_symbol_name_string(name).unwrap();
```

Comment:
This was part of an annoying and hard-to-debug tangent. The presence of `dbg!` statements was sometimes causing spurious errors related to symbol-handling. Some `Str`s needed to get `Sym` hashes were missing from the immutable `Store`.